py_ecc/bls/ciphersuites.py

```
@@ -123,72 +134,76 @@ def Verify(cls, PK: BLSPubkey, message: bytes, signature: BLSSignature) -> bool:
+        return cls._CoreAggregateVerify(PKs, messages, signature, cls.DST)
 
 
 class G2ProofOfPossession(BaseG2Ciphersuite):
-    DST = b'BLS_SIG_BLS12381G2-SHA256-SSWU-RO-_POP_'
-    POP_TAG = b'BLS_POP_BLS12381G2-SHA256-SSWU-RO-_POP_'
+    DST = b'BLS_SIG_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_'
+    POP_TAG = b'BLS_POP_BLS12381G2_XMD:SHA-256_SSWU_RO_POP_'
 
     @classmethod
-    def AggregateVerify(cls, pairs: Sequence[Tuple[BLSPubkey, bytes]],
-                        signature: BLSSignature) -> bool:
-        return cls._CoreAggregateVerify(pairs, signature, cls.DST)
+    def AggregateVerify(cls, PKs: Sequence[BLSPubkey],
+                        messages: Sequence[bytes], signature: BLSSignature) -> bool:
+        return cls._CoreAggregateVerify(PKs, messages, signature, cls.DST)
 
     @classmethod
     def PopProve(cls, SK: int) -> BLSSignature:
-        pubkey = cls.PrivToPub(SK)
+        pubkey = cls.SkToPk(SK)
         return cls._CoreSign(SK, pubkey, cls.POP_TAG)
 
     @classmethod
     def PopVerify(cls, PK: BLSPubkey, proof: BLSSignature) -> bool:
         return cls._CoreVerify(PK, PK, proof, cls.POP_TAG)
 
     @staticmethod
-    def _AggregatePKs(PKs: Sequence[BLSPubkey]) -> BLSPubkey:
-        accumulator = Z1  # Seed with the point at infinity
-        for pk in PKs:
-            pubkey_point = pubkey_to_G1(pk)
-            accumulator = add(accumulator, pubkey_point)
-        return G1_to_pubkey(accumulator)
+    def _AggregatePKs(PKs: Sequence[BLSPubkey]) -> Union[BLSPubkey, bool]:
+        try:
+            if len(PKs) < 1:
+                raise ValidationError('Insufficient number of PKs. (n < 1)')
+            accumulator = Z1  # Seed with the point at infinity
+            for pk in PKs:
+                pubkey_point = pubkey_to_G1(pk)
+                accumulator = add(accumulator, pubkey_point)
+            return G1_to_pubkey(accumulator)
+        except ValidationError:
+            return False
```

Same error handling, IMO it's better to throw the exception.